kubemarine/core/cluster.py

```diff
@@ -498,13 +498,17 @@ def check_nodes_accessibility(self, skip_check_iaas: bool = True) -> None:
         procedure: str = self.context['initial_procedure']
         if procedure == 'check_iaas' and skip_check_iaas:
             return
+        
+        if procedure == 'remove_node':
+            group = self.make_group_from_roles(['control-plane', 'balancer'])
+        else:
+            group = self.make_group_from_roles(['control-plane', 'balancer']).include_group(self.get_new_nodes_or_self())
 
-        # Check that only subset of nodes for removal can be offline
-        remained_offline = self.nodes['all'].get_online_nodes(False)
+        # Check that nodes are online
+        remained_offline = group.get_online_nodes(False)
 
-        # Check that all online nodes are accessible.
-        all_nodes = self._get_all_nodes()
-        inaccessible_online = all_nodes.get_online_nodes(True).exclude_group(all_nodes.get_accessible_nodes())
+        # Check that nodes are accessible.
+        inaccessible_online = group.get_online_nodes(True).exclude_group(group.get_accessible_nodes())
```

Comment:
I've added to check `get_changed_nodes()`.  Thus, access to the removed nodes will also be checked if they are online.
I consider this check unnecessary for existing workers. Just like for remote nodes, if the laptop is running and connected to the network, it will be online even if the VM with the node is not running.